packages/core/src/agent/index.ts

```diff
@@ -646,7 +646,15 @@ export class Agent<
         let coreMessages = messages;
         let threadIdToUse = threadId;
 
-        if (this.getMemory() && resourceId) {
+        const memory = this.getMemory();
+
+        if (threadId && memory && !resourceId) {
```

Comment:
shouldn't we also update the types?

Reply:
Good call, updated in https://github.com/mastra-ai/mastra/pull/2602/commits/e05d239dc37cfed12170529912d73d1877ffc3af